src/formats/AmberNetCDF.cpp

```diff
@@ -97,15 +98,26 @@ AmberNetCDFFormat::AmberNetCDFFormat(std::string path, File::Mode mode, File::Co
     }
 }
 
-size_t AmberNetCDFFormat::nsteps() {
+template <>
+size_t Amber<NCRST>::nsteps() {
+    return 1;
+}
+
+template <>
+size_t Amber<NCTRJ>::nsteps() {
     return file_.dimension("frame");
 }
 
-void AmberNetCDFFormat::read_step(const size_t step, Frame& frame) {
+template <AmberFormat F>
+void Amber<F>::read_step(const size_t step, Frame& frame) {
```

Comment:
this should check for `step != 0 && F == NCRST` as well, since the user could call this function directly.

Reply:
The check should have been here in the first place. Moved to this position now.